lxd/firewall/interfaces.go

```diff
@@ -0,0 +1,32 @@
+package firewall
+
+import (
+	"net"
+
+	deviceConfig "github.com/lxc/lxd/lxd/device/config"
+)
+
+
+// Firewall represents an LXD firewall.
+type Firewall interface {
+	// Lower-level Functions
+	NetworkClear(protocol string, comment string, table string) error
+	ContainerClear(protocol string, comment string, table string) error
```

Comment:
So I think it would make sense to abstract two a bit, so rather than passing the comment string, instead have the Firewall logic be responsible for generating that string.

That would likely make those functions:
```
NetworkClear(name string, protocol string, table string) error
InstanceClear(name string, project string, protocol string, table string) error
```

And alternative for that latter would be:
```
InstanceClear(inst Instance, protocol string, table string) error
```

This option would be better if we know that the function is called in contexts where we do have an existing Instance struct to pass (so typically anywhere from within container_lxc or vm_qemu).

Also note that the above changes it from Container to Instance as we're slowly converting the entire code base to that naming which covers both containers and virtual machines.

Reply:
This has been resolved in a recent commit.